cdap-proto/src/main/java/co/cask/cdap/proto/id/ApplicationId.java

```diff
@@ -105,7 +107,10 @@ public static ApplicationId fromIdParts(Iterable<String> idString) {
 
   @Override
   protected Iterable<String> toIdParts() {
-    return ImmutableList.of(namespace, application);
+    List<String> parts = new ArrayList<>();
+    parts.add(namespace);
+    parts.add(application);
+    return Collections.unmodifiableList(parts);
```

Comment:
There is a one line equivalent of this:

```java
return Collections.unmodifiableList(Arrays.asList(new String[] { namespace, application }));
```

You don't have to change it if you don't want to. Just a note.